syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.routeguide";
option java_outer_classname = "RouteGuideProto";
option objc_class_prefix = "RTG";

package chat;

message Vacio {}

message NotaInicial {
	string nombre = 1 
}

message Nota {
    string ID = 1;
    string mensaje = 2;
}

service ChatServer {
    
    //servicio que permite mantener el streaming del chat asincrono activo entre 2 personas
    rpc ChatStream (Vacio) returns (stream Nota);
    //servicio que permite enviar mensajes al servidor que procesara luego
    rpc EnviarNota (Nota) returns (Vacio);
}